workflow/executor/resource.go

```diff
@@ -308,10 +308,15 @@ func (we *WorkflowExecutor) SaveResourceParameters(resourceNamespace string, res
 		log.Info(cmd.Args)
 		out, err := cmd.Output()
 		if err != nil {
-			if exErr, ok := err.(*exec.ExitError); ok {
-				log.Errorf("`%s` stderr:\n%s", cmd.Args, string(exErr.Stderr))
+			// We have a default value to use instead of returning an error
+			if param.ValueFrom.Default != "" {
```

Comment:
test?

Reply:
I didn't include tests for this because it would be difficult to mock this behavior and I have already included a non-E2E test that covers the same behavior here: https://github.com/argoproj/argo/pull/2500/files#diff-4ed679a69dc8fdd934fb1e7f7c0f3259R109-R139

If you think it's necessary, I can invest the time into mocking this or creating an e2e test